crates/bevy_ecs/src/system/function_system.rs

```diff
@@ -56,6 +56,17 @@ impl SystemMeta {
         &self.name
     }
 
+    /// Sets the name of of this system
```

Comment:
The context from the PR description about how this is useful for closure systems with debug tools is really useful, and belongs in the docs for both these methods.